grow/translators/google_sheets.py

```diff
@@ -382,11 +388,83 @@ def _generate_create_sheets_requests(self, catalogs, source_lang):
                 },
             })
 
-            requests += self._generate_comments_column_requests(sheet_id, catalog)
+            requests += self._generate_comments_column_requests(
+                sheet_id, catalog)
+
+        return requests
+
+    def _generate_filter_view_requests(self, sheet_id, sheet, filter_view):
+        requests = []
+
+        is_filtered = False
+        if sheet and 'filterViews' in sheet:
+            for existing_range in sheet['filterViews']:
+                if existing_range['filterViewId'] == filter_view['filterViewId']:
+                    is_filtered = True
+                    requests.append({
+                        'updateFilterView': {
+                            'filter': filter_view,
+                            'fields': 'range,title,criteria',
+                        },
+                    })
+                    break
+
+        if not is_filtered:
+            requests.append({
+                'addFilterView': {
+                    'filter': filter_view,
+                },
+            })
 
         return requests
 
-    def _generate_style_requests(self, sheet_id, sheet=None):
+    def _generate_filter_view_resource_requests(self, sheet_id, sheet, catalog=None):
+        requests = []
+
+        if not catalog:
+            return requests
+
+        lang = str(catalog.locale)
+        location_to_filter_id = {}
+        filter_ids = set()
+
+        # Find all the unique resource locations.
+        for message in catalog:
+            if not message.id:
+                continue
+
+            for raw_location in message.locations:
+                location = raw_location[0]
+                if not location in location_to_filter_id:
+                    # Try to match up with the document hash value.
+                    location_to_filter_id[location] = self._trim_hash(
```

Comment:
Can you move this to a function, something like `generate_filter_id` with args location and lang? Then we can reuse it easily for creating the per-document edit URL.